src/a_star_search/core.clj

```diff
@@ -1,24 +1,33 @@
 (ns a-star-search.core
   (:require [search.algorithms :as alg]
-            [problems.n-puzzle :as np])
+            [problems.n-puzzle :as np]
+            [problems.maze :as mz]
+            [clojure.tools.namespace.repl :refer [refresh]])
   (:gen-class))
 
+(def x "x") ;Pay no attention to the x behind the curtain. This varibale lets us use x without quotes.
+(def test-board [[0 0 0 0 0 0 0 0]
+                 [x x x x 0 x x 0]
+                 [0 x x x 0 x 0 0]
+                 [0 0 0 0 0 x 0 x]
+                 [0 x 0 x 0 0 0 x]])
+
 (defn print-results [result path costs goal-state]
   (println (str "We explored " (count result) " states."))
-  (println "The path to the solution is:")
-  (doseq [b (map :board path)]
+  (println "The path, in [row col] format, to the solution is:")
+  (doseq [b (map :current-pos path)]
     (println b))
   (when costs
     (println "The path has" (count path) "steps and its cost is" (costs goal-state))))
 
 (defn -main
   "Search for a hard-coded N-puzzle target state."
   [& args]
-  (let [start-state (np/->State [[0 1 3] [4 2 5] [7 8 6]] [0 0])
-        goal-state (np/->State [[0 1 2] [3 4 5] [6 7 8]] [0 0])
+  (let [start-state (mz/->State test-board [0 0])
+        goal-state (mz/->State test-board [2 7])
         max-states 1000000
-        costs nil
-        came-from (alg/breadth-first-search np/children max-states start-state goal-state)
+        costs (constantly 1)
+        came-from (alg/breadth-first-search mz/children max-states start-state goal-state)
         ; [came-from costs] (alg/shortest-path np/children (constantly 1)
```

Comment:
Both `shortest-path` calls still refer to `np` instead of `mz`. Did you ever try them? (They worked fine for me after I made that change.)